components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -0,0 +1,346 @@
+ */
+
+import { ContextURL, GitpodServer, WorkspaceCreationResult } from "@gitpod/gitpod-protocol";
+import { SelectAccountPayload } from "@gitpod/gitpod-protocol/lib/auth";
+import { ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { Deferred } from "@gitpod/gitpod-protocol/lib/util/deferred";
+import { useCallback, useState } from "react";
+import { Redirect, useLocation } from "react-router";
+import Modal from "../components/Modal";
+import RepositoryFinder from "../components/RepositoryFinder";
+import SelectIDEComponent from "../components/SelectIDEComponent";
+import SelectWorkspaceClassComponent from "../components/SelectWorkspaceClassComponent";
+import { UsageLimitReachedModal } from "../components/UsageLimitReachedModal";
+import { openAuthorizeWindow } from "../provider-utils";
+import { getGitpodService, gitpodHostUrl } from "../service/service";
+import { LimitReachedOutOfHours, LimitReachedParallelWorkspacesModal } from "../start/CreateWorkspace";
+import { StartWorkspaceOptions } from "../start/start-workspace-options";
+import { StartWorkspaceError } from "../start/StartPage";
+import { useCurrentUser } from "../user-context";
+import { SelectAccountModal } from "../user-settings/SelectAccountModal";
+import Spinner from "../icons/Spinner.svg";
+import { useFeatureFlags } from "../contexts/FeatureFlagContext";
+
+export const useNewCreateWorkspacePage = () => {
+    const { startWithOptions } = useFeatureFlags();
+    const user = useCurrentUser();
+    return startWithOptions || user?.additionalData?.isMigratedToTeamOnlyAttribution;
+};
+
+export function CreateWorkspacePage() {
+    const user = useCurrentUser();
+    const location = useLocation();
+    const props = StartWorkspaceOptions.parseSearchParams(location.search);
+
+    const [useLatestIde, setUseLatestIde] = useState<boolean | undefined>(
+        props.ideSettings?.useLatestVersion !== undefined
+            ? props.ideSettings.useLatestVersion
+            : !!user?.additionalData?.ideSettings?.useLatestVersion,
+    );
+    const [selectedIde, setSelectedIde] = useState(
+        props.ideSettings?.defaultIde !== undefined
+            ? props.ideSettings.defaultIde
+            : user?.additionalData?.ideSettings?.defaultIde,
+    );
+    const [selectedWsClass, setSelectedWsClass] = useState<string | undefined>(props.workspaceClass);
+    const [errorWsClass, setErrorWsClass] = useState<string | undefined>(undefined);
+    const [repo, setRepo] = useState<string | undefined>(location.hash.substring(1));
+    const onSelectEditorChange = useCallback(
+        (ide: string, useLatest: boolean) => {
+            setSelectedIde(ide);
+            setUseLatestIde(useLatest);
+        },
+        [setSelectedIde, setUseLatestIde],
+    );
+    const [errorIde, setErrorIde] = useState<string | undefined>(undefined);
+    const [creating, setCreating] = useState<boolean>(false);
```

Comment:
nit: can drop the `<boolean>` type here - it will be inferred from the `false` default passed in.